File: Solutions/1480-RunningSumOf1DArray/RunningSum.ts
function runningSum(nums: number[]): number[] {
    let sum: number = 0;

    for (let i: number = 0; i < nums.length; i++) {
        sum += nums[i];
        nums[i] = sum;
    }

    return nums;
}

// Time complexity  = 0(n);
// Space complexity = 0(1)

File: Practice/RunningSum/RunningSum.java
class Solution {
    public int[] runningSum(int[] nums) {
        int runningTotal = 0;

        for(int i = 0; i < nums.length; i++){
            nums[i] = nums[i] + runningTotal;

            runningTotal += nums[i] - runningTotal;
        };

        return nums; 
    }
    
}